src/gui/labeling/qgslabelinggui.cpp

```diff
@@ -320,19 +347,20 @@ void QgsLabelingGui::setLayer( QgsMapLayer *mapLayer )
 
   setEnabled( true );
 
-  QgsVectorLayer *layer = static_cast<QgsVectorLayer *>( mapLayer );
-  mLayer = layer;
+  mLayer = mapLayer;
+  QgsVectorLayer *vLayer = qobject_cast<QgsVectorLayer *>( mapLayer );
 
-  mTextFormatsListWidget->setLayerType( mLayer ? mLayer->geometryType() : mGeomType );
-  mBackgroundMarkerSymbolButton->setLayer( mLayer );
-  mBackgroundFillSymbolButton->setLayer( mLayer );
+  mTextFormatsListWidget->setLayerType( vLayer ? vLayer->geometryType() : mGeomType );
+  mBackgroundMarkerSymbolButton->setLayer( vLayer );
+  mBackgroundFillSymbolButton->setLayer( vLayer );
 
   // load labeling settings from layer
-  updateGeometryTypeBasedWidgets();
+  if ( vLayer )
+    updateGeometryTypeBasedWidgets();
```

Comment:
I think this will need calling regardless -- there's logic in there based on mGeomType